apis/run/v1alpha3/kappcontrollerconfig_types.go

```diff
@@ -0,0 +1,125 @@
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=tanzu-package-repo-global
+	GlobalNamespace string `json:"globalNamespace,omitempty"`
+
+	Deployment KappDeployment `json:"deployment,omitempty"`
+
+	Config KappConfig `json:"config,omitempty"`
+}
+
+type KappDeployment struct {
+	// Whether to enable host networking for kapp-controller deployment
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=false
+	HostNetwork bool `json:"hostNetwork,omitempty"`
+
+	// The priority value that various system components use to find the priority of the kapp-controller pod
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=system-cluster-critical
+	PriorityClassName string `json:"priorityClassName,omitempty"`
+
+	// Concurrency of kapp-controller deployment
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=4
+	Concurrency int `json:"concurrency,omitempty"`
+
+	// kapp-controller deployment tolerations
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:={{key: CriticalAddonsOnly, operator: Exists}, {effect: NoSchedule, key: node-role.kubernetes.io/master}, {effect: NoSchedule, key: node.kubernetes.io/not-ready}, {effect: NoSchedule, key: node.cloudprovider.kubernetes.io/uninitialized, value: "true"}}
+	Tolerations []map[string]string `json:"tolerations,omitempty"`
+
+	// Bind port for kapp-controller API
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=10350
+	APIPort int `json:"apiPort,omitempty"`
+
+	// Address for metrics server
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=:8080
+	MetricsBindAddress string `json:"metricsBindAddress,omitempty"`
+}
+
+type KappConfig struct {
+	// A cert chain of trusted CA certs. These will be added to the system-wide cert pool of trusted CA's
+	// +kubebuilder:validation:Optional
+	CaCerts string `json:"caCerts,omitempty"`
+
+	// The url/ip of a proxy for kapp controller to use when making network requests
+	// +kubebuilder:validation:Optional
+	HTTPProxy string `json:"httpProxy,omitempty"`
+
+	// The url/ip of a TLS capable proxy for kapp-controller to use when making network requests
+	// +kubebuilder:validation:Optional
+	HTTPSProxy string `json:"httpsProxy,omitempty"`
+
+	// A comma delimited list of domain names which kapp-controller should bypass the proxy for when making requests
+	// +kubebuilder:validation:Optional
+	NoProxy string `json:"noProxy,omitempty"`
+
+	// A comma delimited list of hostnames for which kapp-controller should skip TLS verification
+	// +kubebuilder:validation:Optional
+	DangerousSkipTLSVerify string `json:"dangerousSkipTLSVerify,omitempty"`
+}
+
+// KappControllerConfigStatus defines the observed state of KappControllerConfig
+type KappControllerConfigStatus struct {
+	// Name of the data value secret created by controller
+	// +kubebuilder:validation:Optional
+	SecretRef string `json:"secretRef,omitempty"`
+}
+
+//+kubebuilder:object:root=true
```

Comment:
nitpick: inconsistent spacing `//+kubebuilder` and `// +kubebuilder`

Reply:
Updated to use `//+kubebuilder` everywhere